src/libraries/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -71,17 +71,18 @@ public Socket(SocketInformation socketInformation)
             IPEndPoint ep = new IPEndPoint(tempAddress, 0);
 
             Internals.SocketAddress socketAddress = IPEndPointExtensions.Serialize(ep);
+            int size = socketAddress.Size;
```

Comment:
It'd be more obvious that this isn't leading to a buffer overflow if size was initialized from the same span that's being used to initialize bufferPtr

Reply:
updated for readability. It should not be possible to set Size bigger than buffer but it is not obvious here.